src/logcollector/read_win_event_channel.c

```diff
@@ -522,12 +523,22 @@ DWORD WINAPI event_channel_callback(EVT_SUBSCRIBE_NOTIFY_ACTION action, os_chann
 {
     if (action == EvtSubscribeActionDeliver) {
         send_channel_event(evt, channel);
+    } else {
+        for (int i = 0; i < 3; i++) {
```

Comment:
Another inconsistency between the behaviour between the eventlog and eventchannel modes is that when the service stops, eventlog waits infinitely until it reconnects (thanks to this PR). However, the behaviour in the eventchannel mode is that it never reconnects if the service does not restart in 15 seconds.